CRM/Core/BAO/File.php

```diff
@@ -673,33 +673,20 @@ public static function deleteAttachment() {
       self::deleteEntityFile($params['entityTable'], $params['entityID'], NULL, $params['fileID']);
       return;
     }
-    $refCount = 0;
     // Custom file field - set the custom value to NULL
     $customGroup = CRM_Core_BAO_CustomGroup::getGroup(['table_name' => $params['entityTable']]);
     $customField = $customGroup['fields'][$params['customField']] ?? NULL;
     if ($customField) {
-      // *SIGH* Api4 has a bug which cannot update file custom fields to NULL :(
-      //   $entityName = $customGroup['extends'] ?? NULL;
-      //   $fieldName = CRM_Core_BAO_CustomField::getLongNameFromShortName('custom_' . $params['customField']);
-      //   civicrm_api4($entityName, 'update', [
-      //     'values' => [$fieldName => NULL, 'id' => $params['entityID']],
-      //     'where' => [
-      //       ['id', '=', $params['entityID']],
-      //       [$fieldName, '=', $params['fileID']],
-      //     ],
-      //   ]);
-      // TEMP HACK
-      CRM_Core_DAO::executeQuery("UPDATE `{$customGroup['table_name']}` SET {$customField['column_name']} = NULL WHERE entity_id = %1 AND {$customField['column_name']} = %2", [
-        1 => [$params['entityID'], 'Integer'],
-        2 => [$params['fileID'], 'Integer'],
+      // Api4 will also delete the file from disk if there are no other references to it
+      $entityName = $customGroup['extends'] ?? NULL;
+      $fieldName = CRM_Core_BAO_CustomField::getLongNameFromShortName('custom_' . $params['customField']);
+      civicrm_api4($entityName, 'update', [
+        'values' => [$fieldName => NULL, 'id' => $params['entityID']],
+        'where' => [
+          ['id', '=', $params['entityID']],
+          [$fieldName, '=', $params['fileID']],
+        ],
       ]);
-      $refCount = CoreUtil::getRefCountTotal('File', $params['fileID']);
-    }
-    // Delete file if there are no other references
-    if ($refCount === 0) {
-      \Civi\Api4\File::delete(FALSE)
-        ->addWhere('id', '=', $params['fileID'])
-        ->execute();
```

Comment:
What if it's not a custom field - then this wouldn't run?

Reply:
@demeritcowboy that's true

Reply:
@demeritcowboy if it's not a custom field then it's handled by `deleteEntityFile()` above.

Reply:
Ok I see. So it was just oddly outside the `if` before but was never hit anyway for non-custom.